### PR TITLE
Android geolocation with permissions

### PR DESCRIPTION
**Pull request by jwerle**

### CODE REVIEW

File: api/internal/geolocation.js [hunk 1]
```diff
@@ -84,24 +104,70 @@ function createGeolocationPosition (data) {
...
+  getCurrentPosition: getPlatformFunction('getCurrentPosition'),
+  watchPosition: getPlatformFunction('watchPosition'),
+  clearWatch: getPlatformFunction('clearWatch')
 }
 
-export async function getCurrentPosition (onSuccess, onError, options = {}) {
+/**
+ * @param {function(GeolocationPosition)} onSuccess
+ * @param {onError(Error)} onError
+ * @param {?({ timeout?: number })} [options]
```
Comment (chicoxyzzy) [inline api/internal/geolocation.js:130]: not sure if that will parse correctly

Comment (jwerle) [reply api/internal/geolocation.js:130]: what should it be? 🙏🏼

Comment (chicoxyzzy) [reply api/internal/geolocation.js:130]:
```js
* @param {Object=} options
* @param {number=} options.timeout
```

File: src/core/preload.cc [hunk 2]
```diff
@@ -127,22 +120,37 @@ namespace SSC {
...
+      "                                                                      \n"
+      "  try {                                                               \n"
+      "    const event = '__runtime_init__';                                 \n"
+      "    let onload = null                                                 \n"
+      "    Object.defineProperty(globalThis, 'onload', {                     \n"
+      "      get: () => onload,                                              \n"
+      "      set (value) {                                                   \n"
+      "        const opts = { once: true };                                  \n"
+      "        if (onload) {                                                 \n"
+      "          globalThis.removeEventListener(event, onload, opts);        \n"
```
Comment (chicoxyzzy) [inline src/core/preload.cc:132]: we probably don't need to remove it as it has `{ once: true }` already, but this is fine

File: src/window/apple.mm [hunk 3]
```diff
@@ -681,55 +683,95 @@ -                       (void) webView: (WKWebView*) webView
...
+    } @catch (NSException *error) {
+      debug("Failed to set preference: 'storageAPIEnabled': %@", error);
     }
 
-    if (userConfig["permissions_allow_device_orientation"] == "false") {
-      [prefs setValue: @NO forKey: @"deviceOrientationEventEnabled"];
-    } else {
-      [prefs setValue: @YES forKey: @"deviceOrientationEventEnabled"];
+    @try {
+      if (userConfig["permissions_allow_device_orientation"] == "false") {
```
Comment (chicoxyzzy) [inline src/window/apple.mm:707]: do want want to add this to config?

**Pull request merged**